### ISSUE TITLE
advancement not triggered when enter nether and end portals

### ISSUE THREAD

**Issue by benwoo1110** [Multiverse/Multiverse-NetherPortals#184]
### Information

* **Server version:** tested on 1.14, 1.15 and 1.16

* **Full output of `/mv version -b`:** https://j.mp/30BPzZM

### Details
I was **`able `** to reproduce my issue on a freshly setup and up-to-date server with the latest version of Multiverse plugins with no other plugins and with no kinds of other server or client mods.

**Description**  
"We Need to Go Deeper" advancement not triggered when enter Nether Portal as well as "The End?" for end portals, when portals are made on the non-default world. Probably the way mv-np handles nether portals not compatible with vanilla advancement checking. My draft solution is to manually add to player on EntityEnterPortalEvent.

**Comment by nicegamer7** [comment]
I think this should be handled by the server, not by us.

**Comment by benwoo1110** [comment]
Thinking of it yea, maybe we should contact paper/spigot on it.

**Comment by nicegamer7** [comment]
Yeah, I'm about to take a look and see how easy it is, then gonna report to Spigot.

**Comment by nicegamer7** [comment]
Looks like it's handled by the client :/

I guess it's up to us.

**Comment by benwoo1110** [comment]
> edit: Should be easy though. I'll open a PR tomorrow, or if you wanna do it go ahead.

Haha didn't see your edit, so when ahead and opened a PR for it already.

**Comment by nicegamer7** [comment]
No worries, I'll take a look.

**Comment by nicegamer7** [comment]
It turns out the server CAN trigger these. I'm gonna open a ticket with Spigot.

**Comment by nicegamer7** [comment]
My memory is really bad... did I end up finishing this?

**Comment by nicegamer7** [comment]
I did not. Opening a PR now.

**Comment by benwoo1110** [comment]
Didn’t I fix it here? https://github.com/Multiverse/Multiverse-NetherPortals/pull/218

**Comment by nicegamer7** [comment]
Yeah, but it's cleaner for the server to handle this, so I submitted a PR to Spigot. If it gets merged, then in MV5 we can remove the check in MVNP.

**Comment by benwoo1110** [comment]
Ah, it may be worth checking whether other nether related advancement are not triggered as well? There is one called “Subspace Bubble”, not sure if this also will be not detected, since entering nether portal isn’t detected.

“”Use the Nether to travel between 2 points in the Overworld with a minimum horizontal distance of 7000 blocks between each other, 875 blocks in the Nether.“”

**Comment by nicegamer7** [comment]
Let me see if that's fixed lol.

**Comment by nicegamer7** [comment]
Without either of our changes, unsurprisingly, it doesn't work.

With only your change, it doesn't work.

With only the server change I made, it works :D

**Comment by benwoo1110** [comment]
👍

**Comment by nicegamer7** [comment]
Okay, the PR went through, so this is no longer necessary for us to do in 1.16.5+.

I think it's okay to keep the stuff you added in #218, but we should make it only for versions before 1.16.5.

**Comment by zax71** [comment]
@benwoo1110 should this still be open? Seems like the PR got merged

**Comment by benwoo1110** [comment]
Removing the custom handling in v5.1 as it should be a server side trigger and we target 1.18.2+ now so there is no issues anymore.
